aloha-core/src/main/scala/com/eharmony/aloha/semantics/compiled/plugin/schemabased/SchemaBasedSemanticsPlugin.scala

```diff
@@ -0,0 +1,219 @@
+
+    function.toEither.left.map(l => l.head :: l.tail) // Don't want to rely on scalaz for outward facing APIs.
+  }
+
+  private[schemabased] def generateFunction(p: FieldAccessorPartition) = {
+    val (before, repeated, after) = p
+    val b = determineMappingPartition(before)
+    val a = determineMappingPartition(after)
+    generateFunctionHelper(b, repeated, a)
+  }
+
+  /**
+    * Get a list of lines in the function.  This includes function signature but not the name or return type.
+    * @param before
+    * @param repeated
+    * @param after
+    * @return
+    */
+  private[schemabased] def generateFunctionHelper(before: MappingPartition, repeated: Option[Repeated], after: MappingPartition): VariableAccessorCode = {
+    val cg = codeGenerators
+    import cg.{optionalCodeGen, repeatedCodeGen}
+
+    val (bfm, bm, br) = before
+    val (afm, am, ar) = after
+
+    // Generate any flat mapped elements before the appearance of the repeated element.
+    val bfms = (1 to bfm.size).zip(bfm) map { case (i, (r, o)) => cg.containerCodeGen(r, o, i, FLAT_MAP) }
+
+    // Generate the 0 or 1 mapped element before the appearance of the repeated element.
+    val bms = bm map { case (r, o) => cg.containerCodeGen(r, o, bfms.size + 1, MAP) }
+
+    // Generate the 0 or 1 repeated elements.
+    val mapList = Seq(afm, am.toSeq, ar).foldLeft(false)(_ || _.size > 0)
+    val rep = repeated map { r => cg.containerCodeGen(br, r, bfms.size + bms.size + 1, if (mapList) MAP else NONE) }
+
+    // Generate any mapped elements appearing after the repeated element.
+    val afmi = Seq.range(0, afm.size) map { _ + bfms.size + bms.size + rep.size + 1 }
+    val afms = afmi.zip(afm) map { case (i, (r, o)) => cg.containerCodeGen(r, o, i, FLAT_MAP) }
+
+    // Generate the 0 or 1 mapped elements appearing after the repeated element.
+    val ami = Seq.range(0, am.size) map { _ + bfms.size + bms.size + rep.size + afm.size + 1 }
+    val ams = ami.zip(am) map { case (i, (r, o)) => cg.containerCodeGen(r, o, i, if (ar.nonEmpty) MAP else NONE) }
+
+    // Generate the required elements appearing after the repeated element.
+    val ari = bfms.size + bms.size + rep.size + afm.size + ams.size + 1
+    val ars = Option(ar.nonEmpty).filter(identity).map(_ => cg.NoSuffixCodeGen.unit(ar, ari))
+
+    // Assemble all the generated lines of code.  Determine the appropriate number of right parentheses
+    // (number of generated lines minus 1) and add to the last line.  Finally, if the code produces optional
+    // data before the repeated element, then it would result in one of the following types:
+    // Option[Seq[A]] or Option[Seq[Option[A]]].  Because we want to avoid the outermost Option, we map None
+    // to an empty sequence.
+    val r = bfms ++ bms ++ rep ++ afms ++ ams ++ ars
+
+    // The second clause accomodates that repeated fields can be nullable.
+    val optBeforeList = hasOptionalStuff(before) || repeated.fold(false)(r => r.field.nullable)
```

Comment:
I thought about this but `exists` is more difficult to ascertain the desired intent, so I left it as is.